include/camp/array.hpp

```diff
@@ -155,6 +155,16 @@ namespace camp {
          }
       }
 
+      // TODO: Revisit swap implementations to make them match the std library
+      CAMP_HOST_DEVICE constexpr
+      typename std::enable_if<std::is_nothrow_move_constructible<T>::value &&
+                              std::is_nothrow_move_assignable<T>::value>::type
+      swap(array& a) noexcept(std::is_nothrow_swappable<T>::value) {
```

Comment:
is_nothrow_swappable is C++17

Reply:
Right you are. Converting back to a draft until I have time to look at this again.